test/int.jl

```diff
@@ -201,7 +201,13 @@ end
         end
         for T2 in Base.BitInteger_types
             for op in (>>, <<, >>>)
-                @test Core.Compiler.is_total(Base.infer_effects(op, (T, T2)))
+                if sizeof(T2)==sizeof(Int) || T <: Signed || (op==>>>) || T2 <: Unsigned
+                    @test Core.Compiler.is_total(Base.infer_effects(op, (T, T2)))
+                else
+                    @test Core.Compiler.is_foldable(Base.infer_effects(op, (T, T2)))
+                    # nothrow analysis of recursion needs fixing
+                    @test_broken Core.Compiler.is_total(Base.infer_effects(op, (T, T2)))
```

Comment:
Don't use `:total` but use as accurate effect as possible
```suggestion
                    # #47835, TODO implement interval arithmetic analysis
                    @test_broken Core.Compiler.is_nothrow(Base.infer_effects(op, (T, T2)))
```